src/snowflake/snowpark/stored_procedure_profiler.py

```diff
@@ -0,0 +1,120 @@
+        able to set session-level parameters.
+    """
+
+    def __init__(
+        self,
+        session: "snowflake.snowpark.Session" = None,
+    ) -> None:
+        self._session = session
+        self._query_history = session.query_history(include_thread_id=True)
+
+    def register_modules(self, stored_procedures: List[str]):
+        """
+        Register stored procedures to generate profiles for them.
+
+        Note:
+            Registered modules will be overwritten by this function. Use this function with an empty string will remove
+            registered modules.
+        Args:
+            stored_procedures: List of names of stored procedures.
+        """
+        sql_statement = (
+            f"alter session set python_profiler_modules='{','.join(stored_procedures)}'"
+        )
+        self._session.sql(sql_statement).collect()
+
+    def set_targeted_stage(self, stage: str):
+        """
+        Set targeted stage for profiler output.
+
+        Note:
+            The stage name must be a fully qualified name.
+
+        Args:
+            stage: String of fully qualified name of targeted stage
+        """
+        validate_object_name(stage)
+        if (
+            len(self._session.sql(f"show stages like '{stage}'").collect()) == 0
+            and len(
+                self._session.sql(
+                    f"show stages like '{stage.split('.')[-1]}'"
+                ).collect()
+            )
+            == 0
+        ):
+            self._session.sql(
+                f"create temp stage if not exists {stage} FILE_FORMAT = (RECORD_DELIMITER = NONE FIELD_DELIMITER = NONE )"
+            ).collect()
+        sql_statement = f'alter session set PYTHON_PROFILER_TARGET_STAGE ="{stage}"'
+        self._session.sql(sql_statement).collect()
+
+    def set_active_profiler(self, active_profiler_type: Literal["LINE", "MEMORY"]):
+        """
+        Set active profiler.
+
+        Note:
+            Active profiler must be either 'LINE' or 'MEMORY' (case-sensitive). Active profiler is 'LINE' by default.
+        Args:
+            active_profiler_type: String that represent active_profiler, must be either 'LINE' or 'MEMORY'
+            (case-sensitive).
+
+        """
+        if active_profiler_type not in ["LINE", "MEMORY"]:
+            raise ValueError(
+                f"active_profiler expect 'LINE', 'MEMORY', got {active_profiler_type} instead"
+            )
+        sql_statement = f"alter session set ACTIVE_PYTHON_PROFILER = '{active_profiler_type.upper()}'"
+        self._session.sql(sql_statement).collect()
+
+    def disable(self):
+        """
+        Disable profiler.
+        """
+        sql_statement = "alter session set ACTIVE_PYTHON_PROFILER = ''"
+        self._session.sql(sql_statement).collect()
+
+    @staticmethod
+    def _is_sp_call(query):
+        return re.match(STORED_PROCEDURE_CALL_PATTERN, query, re.DOTALL) is not None
```

Comment:
seems like it only detects anonymous sp calls. can you update the variable name accordingly.

Reply:
I'll merge them